src/analysis/complex/locally_uniform_limit.lean

```diff
@@ -0,0 +1,167 @@
+
+## Main results
+
+* `tendsto_locally_uniformly_on.differentiable_on`: A locally uniform limit of holomorphic functions
+  is holomorphic.
+* `tendsto_locally_uniformly_on.deriv`: Locally uniform convergence implies locally uniform
+  convergence of the derivatives to the derivative of the limit.
+-/
+
+open set metric measure_theory filter complex interval_integral
+open_locale real topological_space
+
+variables {E ι : Type*} [normed_add_comm_group E] [normed_space ℂ E] [complete_space E]
+  {U K : set ℂ} {z : ℂ} {M r δ : ℝ} {φ : filter ι} {F : ι → ℂ → E} {f g : ℂ → E}
+
+namespace complex
+
+section cderiv
+
+/-- A circle integral which coincides with `deriv f z` whenever one can apply the Cauchy formula for
+the derivative. It is useful in the proof of the Weierstrass theorem because it depends continuously
+on `f` for the uniform topology. -/
+noncomputable def cderiv (r : ℝ) (f : ℂ → E) (z : ℂ) : E :=
+  (2 * π * I : ℂ)⁻¹ • ∮ w in C(z, r), ((w - z) ^ 2)⁻¹ • f w
+
+lemma cderiv_eq_deriv (hU : is_open U) (hf : differentiable_on ℂ f U) (hr : 0 < r)
+  (hzr : closed_ball z r ⊆ U) :
+  cderiv r f z = deriv f z :=
+two_pi_I_inv_smul_circle_integral_sub_inv_smul_of_differentiable hU hzr hf (mem_ball_self hr)
+
+lemma norm_cderiv_le (hr : 0 < r) (hf : ∀ w ∈ sphere z r, ‖f w‖ ≤ M) :
+  ‖cderiv r f z‖ ≤ M / r :=
+begin
+  have hM : 0 ≤ M,
+  { obtain ⟨w, hw⟩ : (sphere z r).nonempty := normed_space.sphere_nonempty.mpr hr.le,
+    exact (norm_nonneg _).trans (hf w hw) },
+  have h1 : ∀ w ∈ sphere z r, ‖((w - z) ^ 2)⁻¹ • f w‖ ≤ M / r ^ 2,
+  { intros w hw,
+    simp only [mem_sphere_iff_norm, norm_eq_abs] at hw,
+    simp only [norm_smul, inv_mul_eq_div, hw, norm_eq_abs, map_inv₀, complex.abs_pow],
+    exact div_le_div hM (hf w hw) (sq_pos_of_pos hr) le_rfl },
+  have h2 := circle_integral.norm_integral_le_of_norm_le_const hr.le h1,
+  simp only [cderiv, norm_smul],
+  refine (mul_le_mul le_rfl h2 (norm_nonneg _) (norm_nonneg _)).trans (le_of_eq _),
+  field_simp [_root_.abs_of_nonneg real.pi_pos.le, real.pi_pos.ne.symm, hr.ne.symm],
+  ring
+end
+
+lemma cderiv_sub (hr : 0 < r) (hzr : closed_ball z r ⊆ U)
+  (hf : continuous_on f U) (hg : continuous_on g U) :
+  cderiv r (f - g) z = cderiv r f z - cderiv r g z :=
+begin
+  have h1 : continuous_on (λ (w : ℂ), ((w - z) ^ 2)⁻¹) (sphere z r),
+  { refine ((continuous_id'.sub continuous_const).pow 2).continuous_on.inv₀ (λ w hw h, hr.ne _),
+    rwa [mem_sphere_iff_norm, sq_eq_zero_iff.mp h, norm_zero] at hw },
+  simp_rw [cderiv, ← smul_sub],
+  congr' 1,
+  simp_rw [pi.sub_apply, smul_sub],
+  refine circle_integral.integral_sub _ _,
+  { have := (h1.smul (continuous_on.mono hf ((sphere_subset_closed_ball).trans hzr))),
+    exact continuous_on.circle_integrable hr.le this },
+  { have := (h1.smul (continuous_on.mono hg ((sphere_subset_closed_ball).trans hzr))),
+    exact continuous_on.circle_integrable hr.le this }
+end
+
+lemma norm_cderiv_sub_lt (hr : 0 < r) (hzr : closed_ball z r ⊆ U)
```

Comment:
Could you split this into first `norm_cderiv_lt` (for one single function) and then the `sub` version?